api/docs/mapstruct.md

```diff
@@ -0,0 +1,608 @@
+# MapStruct Best Practices
+## What is MapStruct?
+[MapStruct](https://mapstruct.org/documentation/stable/reference/html/) is a Java code generator specializing in writing
+conversion or mapping functions between an object instance and an instance of
+a corresponding object in a class designed for another architectural tier, protocol, or technology.
+It works mainly by matching types and names via reflection, or using programmer-provided
+hints where that strategy isn't effective.
+## Is this a gimmick? Why do I need it?
+Everything MapStruct does is simple and deterministic, and every programmer has done it before. So why
+bring in another library and codegen step? What's the advantage?
+
+First, by defining a mapper, you're adding some executable documentation to your types, and being
+explicit about how they should map.
+ 
+Second, you don't have to rely on developers to perform awkward null
+checks for every field, as you get that for free. The generated code is very uniform and thus
+quite readable.
+
+Third, and most important, is that once you've defined a mapper, maintenance is very easy. New fields
+will simply be mapped for you (assuming they exist on both the source and the target type). If there's
+a new target field that doesn't have a sourcce, youll gget a compile-time warning reminding you. (It's
+important to stay on top of those). 
+
+### Examples
```

Comment:
I would prefer to have this section at the very end of the doc or possibly in a subdirectory for stories and other protracted examples. It distracts from the actual reason I come looking for this file - the coding standards we want to uphold when using this library.

Reply:
I think my preference here is to put the examples where they are relevant. A full, complete example should go at the bottom (I agree with Ariel) and maybe have a link here. But I would love to see examples scattered throughout with each standard.

Reply:
Pulling it out into its own tutorial makes sense.